core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -64,27 +64,354 @@
+            cp->right = query::ValueExpr::newSimple(
+                           query::ValueFactor::newConstFactor("1"));
+            terms.push_back(cp);
+            return newFactor;
+        }
+
+        char const* const fName;
+        int const paramCount;
+        StringVector const& params;
+        static const int USE_STRING = -999;
+    };
+
+    void _setGenerator(query::QsRestrictor const& r) {
+        if (r._name == "qserv_areaspec_box") {
+            _generator = std::make_shared<AreaGenerator>("s2PtInBox",
+                                                         4,
+                                                         r._params
+                                                         );
+        } else if (r._name == "qserv_areaspec_circle") {
+            _generator = std::make_shared<AreaGenerator>("s2PtInCircle",
+                                                         3,
+                                                         r._params
+                                                         );
+        } else if (r._name == "qserv_areaspec_ellipse") {
+            _generator = std::make_shared<AreaGenerator>("s2PtInEllipse",
+                                                         5,
+                                                         r._params
+                                                         );
+        } else if (r._name == "qserv_areaspec_poly") {
+            const int use_string = AreaGenerator::USE_STRING;
+            _generator = std::make_shared<AreaGenerator>("s2PtInCPoly",
+                                                         use_string,
+                                                         r._params
+                                                         );
+        } else {
+            throw qana::AnalysisBug("Unmatched restriction spec: " + _name);
+        }
+    }
+    std::string _name;
+    std::vector<double> _params;
+    std::shared_ptr<Generator> _generator;
+};
+
+
+query::BoolTerm::Ptr makeCondition(std::shared_ptr<const query::QsRestrictor> const restr,
+                                   RestrictorEntry const& restrictorEntry) {
+    Restriction r(*restr);
+    return r.generate(restrictorEntry);
+}
+
+
+/**
+ * @brief Determine if the given ValueExpr represents a func that is one of the scisql functions that
+ *        starts with `scisql_s2PtIn` that represents an area restrictor.
+ *
+ * This is a helper function for handleScisqlRestrictors.
+ *
+ * @param valueExpr The ValueExpr to check.
+ * @return true if the ValueExpr is a function and starts with `scisql_s2PtIn` else false.
+ */
+bool isScisqlAreaFunc(query::ValueExpr const& valueExpr) {
+    if (not valueExpr.isFunction()) return false;
+    auto&& funcExpr = valueExpr.getFunction();
+    if (funcExpr->getName().find("scisql_s2Pt") != 0) return false;
+    return true;
+}
+
+
+/**
+ * @brief If there is exactly one scisql area restrictor in the top level AND of the where clause,
```

Comment:
I'm not sure the implementation of the method finds "exactly one" or just the very first restrictor(?)  Is it guaranteed that the the top level ("root") term has zero or 1 "bool factor" term?

Reply:
it's exactly one; see ```if (scisqlFunc != nullptr) {
                        return nullptr;```